client_delivery/shared/src/commonMain/kotlin/presentation/login/LoginScreenInteractionListener.kt

```diff
@@ -0,0 +1,20 @@
+package presentation.login
+
+import presentation.base.BaseInteractionListener
+
+interface LoginScreenInteractionListener: BaseInteractionListener {
+    fun onUserNameChanged(userName: String)
+    fun onPasswordChanged(password: String)
+    fun onKeepLoggedInClicked()
+    fun onClickLogin(userName: String, password: String, isKeepMeLoggedInChecked: Boolean)
+
+    //region permission
+    fun onOwnerEmailChanged(ownerEmail: String)
+    fun onRestaurantNameChanged(restaurantName: String)
+    fun onDescriptionChanged(description: String)
+    fun onRequestPermissionClicked()
+    fun onSubmitClicked(restaurantName: String, ownerEmail: String, description: String)
+    fun onCancelClicked()
+    fun onSheetBackgroundClicked()
+    //endregion
+}
```

Comment:
```suggestion
interface LoginScreenInteractionListener: PermissionInteractionListener{
    fun onUserNameChanged(userName: String)
    fun onPasswordChanged(password: String)
    fun onKeepLoggedInClicked()
    fun onClickLogin(userName: String, password: String, isKeepMeLoggedInChecked: Boolean)
}

interface PermissionInteractionListener: BaseInteractionListener{
   fun onOwnerEmailChanged(ownerEmail: String)
    fun onRestaurantNameChanged(restaurantName: String)
    fun onDescriptionChanged(description: String)
    fun onRequestPermissionClicked()
    fun onSubmitClicked(restaurantName: String, ownerEmail: String, description: String)
    fun onCancelClicked()
    fun onSheetBackgroundClicked()
}
```

Reply:
separation of concerns

Reply:
hello @mosayed01, your review is so appreciated, but the BaseScreen take only one listener so we can't do this sepration, also these are one screen as the agreement of uncle Bareq

Reply:
LoginScreenInteractionListener implement PermissionInteractionListener also so we can use LoginScreenInteractionListener as parent interface to our screen model 
The separate will be in permission bottom sheet it will take just PermissionInteractionListener cause no need to know about hole screen interaction

Reply:
thanks so much for your review